OpenTracing/OpenTracing.OpenTracing/ISpan.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+
+namespace OpenTracing
+{
+    public interface ISpan<T> 
+    {
+        T GetSpanContext();
+
+        void Finish();
+        void FinishWithOptions(DateTime finishTime);
+
+        void SetBaggageItem(string restrictedKey, string value);
```

Comment:
My thought to get the baggage was to do something like this.
```csharp
var memoryCarrier = new MemoryTextMapCarrier(new Dictionary<string, string>() { });
tracer.Inject(span, memoryCarrier);            
var value = memoryCarrier.TextMap[key]
```
But I agree, a GetBaggage(key, value) method on the span is more convinient. I will add it.